spec/models/registration_system_spec.rb

```diff
@@ -105,4 +107,134 @@
       expect(RegistrationSystem.verify_credentials).to be_truthy
     end
   end
+
+  describe ".update_rhsm_conf_queue" do
+    before do
+      EvmSpecHelper.create_guid_miq_server_zone
+      @proxy_args = {:registration_http_proxy_server   => "192.0.2.0:myport",
+                     :registration_http_proxy_username => "my_dummy_username",
+                     :registration_http_proxy_password => "my_dummy_password"}
+
+      @miq_region = FactoryGirl.create(:miq_region, :region => 1)
+      allow(MiqRegion).to receive(:my_region).and_return(@miq_region)
+    end
+
+    it "does not modify original arguments" do
+      cloned_creds = @proxy_args.clone
+      RegistrationSystem.update_rhsm_conf_queue(@proxy_args)
+      expect(@proxy_args).to eq(cloned_creds)
+    end
+
+    it "validate that a message was queued for each server in the region" do
+      RegistrationSystem.update_rhsm_conf_queue(@proxy_args).each do |server|
```

Comment:
Would it be ok to ignore the return type from `update_rhsm_conf_queue`?

Do something like:

```
RegistrationSystem.update_rhsm_conf_queue(@proxy_args)
MiqRegion.my_region.miq_servers.each do |server|
```